src/odemis/acq/millmng.py

```diff
@@ -0,0 +1,504 @@
+                                                             1])  # shift in m - absolute position
+                            logging.debug("New Ion-beam shift in m : %s", self._scanner.shift.value)
+                            logging.debug("pixel size in m : %s", pixel_size)
+
+                    else:
+                        if self._future._task_state == CANCELLED:
+                            raise CancelledError()
+
+                        self._future.running_subf = mill_rectangle(milling_settings.roi.value, self._scanner,
+                                                                   iteration=self._iterations[setting_nb],
+                                                                   duration=self._pass_duration[setting_nb],
+                                                                   probe_size=probe_size, overlap=overlap)
+                        try:
+                            self._future.running_subf.result()
+                            logging.debug(f"Milling finished for feature: {site.name.value}")
+                        except Exception as exp:
+                            logging.error(
+                                f"Milling setting: {milling_settings.name.value} for feature: {site.name.value} failed: {exp}")
+                            raise
+
+                    self._scanner.blanker.value = True
+
+                    if self._future._task_state == CANCELLED:
+                        raise CancelledError()
+
+                site.status = self._feature_status
+                logging.debug(f"The milling of feature {site.name.value} completed")
+
+                # Acquire the stream of milled site
+                data = []
+                try:
+                    self._future.running_subf = acquire(self.streams)
+                    data, exp = self._future.running_subf.result()
+                    if exp:
+                        logging.warning(
+                            f"Acquisition for feature {site.name.value} partially failed: {exp}")
+                except Exception as exp:
+                    logging.exception(f"Acquisition for feature {site.name.value} failed: {exp}")
+
+                # check if cancellation happened while the acquiring future is working
+                if self._future._task_state == CANCELLED:
+                    raise CancelledError()
+
+                # Check on the acquired data
+                if not data:
+                    logging.warning("The acquired data array in stream %s for feature %s is empty", self.streams,
+                                    site.name)
+                else:
+                    # Convert the data to StaticStreams, and add them to the feature
+                    new_streams = dataio.data_to_static_streams(data)
+                    site.streams.value.extend(new_streams)
+                    logging.info("The acquisition for stream %s for feature %s is done", self.streams, site.name)
```

Comment:
To keep the function a little shorter, I'd suggest to move this part to a separate function `_acquire_feature(future, feature, streams)`.

Reply:
Used function as  _acquire_feature(self, site:CryoFeature) -> CryoFeature: in the class of the milling task. Would that work?